ompi/runtime/mpiruntime.h

```diff
@@ -55,6 +55,8 @@ OMPI_DECLSPEC extern bool ompi_mpi_initialized;
 OMPI_DECLSPEC extern bool ompi_mpi_finalized;
 /** Has the RTE been initialized? */
 OMPI_DECLSPEC extern bool ompi_rte_initialized;
+/** Did mpi start to finalize? */
+OMPI_DECLSPEC extern int32_t ompi_mpi_finalize_started;
```

Comment:
yep, we did a lot of testing w/ this patch and tried to keep it as close as possible to the original code.
Maybe we can  add type change in master only (for safety) and keep release branches w/ minimal refactoring changes?